### PR TITLE
Auth flow components (TextInput, PasswordInput, Button, Checkbox)

### PR DESCRIPTION
**Pull request by Aphexis**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)
Combo of a couple tickets related to authentication UI work

## Implementation description
* Modified Chakra UI components to fit HUG designs
* Added some props and states based on what's necessary for user auth (e.g. input error states)

## Steps to test
1. Go to /signup-test to see components and test them out

## What should reviewers focus on?
* Any basic changes to the structure of the component?
* This has our current components for TextInput, PasswordInput, AuthButton, CheckboxComp
* There might be some minor changes in PRs for the authentication pages

TODO
- [x] Button disabled state
- [ ] Cleaning up props for components (probably to inherit from the Chakra UI component props)
- [ ] Possibly refactor TextInput and PasswordInput into one component since they're quite similar

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+import { Header, Main, Cards, Footer, AuthButton } from "@components";
 
 const Home: React.FC = () => {
     return (
         <Flex direction="column" minH="100vh">
             <Header />
-            <Main />
+            <Main title="login page" />
             <Cards />
+            <AuthButton text="Hello there!" />
```
Comment (mkanoria) [inline pages/index.tsx:12]: Assuming this was for testing? Can we remove this from the components?

File: pages/signup-test.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { Header, Main, Footer, TextInput, PasswordInput, AuthButton, CheckboxComp } from "@components";
+
+const Home: React.FC = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [emailInvalid, setEmailInvalid] = useState(false);
+    const [passwordInvalid, setPasswordInvalid] = useState(false);
+
+    useEffect(() => {
+        console.log("email", email);
```
Comment (mkanoria) [inline pages/signup-test.tsx:13]: Can we remove console logs?

File: pages/signup-test.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        if (email == "invalid") {
+            setEmailInvalid(true);
+        } else {
+            setEmailInvalid(false);
+        }
+    }, [email]);
+
+    useEffect(() => {
+        console.log("password", password);
```
Comment (mkanoria) [inline pages/signup-test.tsx:22]: ^ same here

File: pages/signup-test.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            </Box>
+            <Spacer />
+            <Footer />
+        </Flex>
+    );
+};
+
+export default Home;
+
+// Adil's signup page:
```
Comment (mkanoria) [inline pages/signup-test.tsx:66]: Let's also remove commented out code?

File: pages/signup.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import { Header, Main, Footer, TextInput, PasswordInput, MyButton, CheckboxComp } from "@components";
+
+const Home: React.FC = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [emailInvalid, setEmailInvalid] = useState(false);
+    const [passwordInvalid, setPasswordInvalid] = useState(false);
+
+    useEffect(() => {
+        console.log("email", email);
```
Comment (mkanoria) [inline pages/signup.tsx:13]: ^ console log

File: pages/signup.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        if (email == "invalid") {
+            setEmailInvalid(true);
+        } else {
+            setEmailInvalid(false);
+        }
+    }, [email]);
+
+    useEffect(() => {
+        console.log("password", password);
```
Comment (mkanoria) [inline pages/signup.tsx:22]: ^

File: pages/signup.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                <PasswordInput
+                    placeholder="*****"
+                    name="password"
+                    label="Password"
+                    errorMessage="Password can't be 1234"
+                    helperText="Must contain one uppercase letter and one lowercase letter"
+                    isInvalid={passwordInvalid}
+                    onChange={(event) => setPassword(event.currentTarget.value)}
+                />
+                <CheckboxComp text="Testing" required={true} isDisabled={false} onChange={() => console.log("HELLO")} />
```
Comment (mkanoria) [inline pages/signup.tsx:55]: Let's also remove test components?

File: pages/signup.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            </Box>
+            <Spacer />
+            <Footer />
+        </Flex>
+    );
+};
+
+export default Home;
+
+// Adil's signup page:
```
Comment (mkanoria) [inline pages/signup.tsx:66]: Same comments as prev file

**Review by mkanoria — CHANGES_REQUESTED** [review]

**Comment by Aphexis** [comment]
Sorry, didn't realize how messy this PR was 😅 

I force pushed with a neater commit. Now this PR only contains the new components and no page changes.
Note: There are still some component updates in future PRs unfortunately (if it's simple I might cherry pick them onto this branch as well), but these components are functional as is.

File: src/components/authButton/index.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                }}
+                backgroundColor="blue.500"
+                textColor="white"
+                fontSize="16px"
+                variant="solid"
+                size="md"
+                height="40px"
+                width="312.01px"
+                borderRadius="0px"
+                fontFamily="Geogrotesque"
```
Comment (yashgajaria) [inline src/components/authButton/index.tsx:30]: Not sure if we still wanted the font to be stated here or was it just going to be globally controlled

**Review by yashgajaria — APPROVED** [review]
LGTM. Other than the minor font issues with the button

File: src/components/authButton/index.tsx (hunk 9 above)
Comment (mkanoria) [reply src/components/authButton/index.tsx:30]: Good catch, we might update this when Tony's changes go in

File: src/components/checkbox/index.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+interface Props {
+    text?: string;
+    value?: boolean;
+    isDisabled?: boolean;
+    required?: boolean;
+    onChange(event: any): any;
+}
+
+export const CheckboxComp: FunctionComponent<Props> = (props: Props) => {
+    const [checkedItem, setCheckedItem] = React.useState(props.value);
```
Comment (mkanoria) [inline src/components/checkbox/index.tsx:20]: Let's just import `useState` directly from react
(Similar to your other PR)

**Review by mkanoria — COMMENTED** [review]
Looks good overall!
2 things - 
1. I would remove the `.DS_Store` files
2. Let's remove the unused imports from the checkbox component

File: src/components/authButton/index.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import React, { FunctionComponent } from "react";
```
Comment (mkanoria) [inline src/components/authButton/index.tsx:1]: We might want to move this into a 'variant' as Tony showed today - might be cleaner. I'm also okay with merging this in rn however and making a bug ticket for this in the future - your call @Aphexis

**Review by mkanoria — APPROVED** [review]
This looks great! Small comment but I'm approving the PR!

**Pull request merged**